### PR TITLE
Add Events API endpoint

### LINKED ISSUE #135: Implement API to load application resource events
Application details page should contains information about selected kubernetes resource. It is useful for troubleshooting ( E.g. kubenetes is unable to create loadbalancer  due to permission issues or cloud account limits)

### PR DESCRIPTION
**Pull request by merenbach**
Closes #135 and addresses concerns about protobuf response object nomenclature.

Test as follows, replacing the resource name and resource uid as required:

```
curl '127.0.0.1:8080/api/v1/applications/guestbook/events?resName=guestbook-ui-5f455cc986-brbrz&resUid=9e0b4491-642a-11e8-81cf-08002774c671'
```

### CODE REVIEW

File: server/application/application.proto [hunk 1]
```diff
@@ -27,6 +34,11 @@ message ManifestQuery {
 
 message ApplicationResponse {}
 
+// ApplicationResourceEventsResponse holds a list of resource events.
+message ApplicationResourceEventsResponse {
+	repeated k8s.io.api.events.v1beta1.Event data = 1;
```
Comment (jessesuen) [inline server/application/application.proto]: Can we name the field `items` and rename this to EventsList? I checked and we are starting to become inconsistent with our response objects in the case of lists. We currently already have ApplicationList, ClusterList, and RepositoryList. I think we should continue that convention:

```
// EventsList holds a list of resource events.
message EventsList {
	repeated k8s.io.api.events.v1beta1.Event items = 1;
}
```

Similarly the repository.ListDirResponse should probably change as well. For consistency sake I think we should change from:

```
// ListDirResponse returns the contents of the repo of a ListDir request
message ListDirResponse {
    repeated string data = 1;
}
```
to:

```
// FileList returns the file names of the repo of a ListDir request
message FileList {
    repeated string items = 1;
}
```

File: server/application/application.go [hunk 2]
```diff
@@ -131,6 +131,11 @@ func (s *Server) Get(ctx context.Context, q *ApplicationQuery) (*appv1.Applicati
 	return s.appclientset.ArgoprojV1alpha1().Applications(s.ns).Get(*q.Name, metav1.GetOptions{})
 }
 
+// ListResourceEvents returns a list of event resources
+func (s *Server) ListResourceEvents(ctx context.Context, q *ApplicationResourceEventsQuery) (*v1.EventList, error) {
+	return s.kubeclientset.CoreV1().Events(s.ns).List(metav1.ListOptions{})
```
Comment (alexmt) [inline server/application/application.go]: I think this will return all events instead of events related to the specified resource. please fix it

**Review by alexmt — CHANGES_REQUESTED** [review]

File: server/application/application.go (hunk 2 above)
Comment (merenbach) [reply server/application/application.go]: @alexmt right you are!  I believe this is now addressed.

File: server/application/application.proto [hunk 3]
```diff
@@ -19,6 +19,12 @@ message ApplicationQuery {
 	optional string name = 1;
 }
 
+// ApplicationEventsQuery is a query for application resource events
+message ApplicationResourceEventsQuery {
+	required string appName = 1;
+	required string resName = 2;
```
Comment (alexmt) [inline server/application/application.proto]: I've realized that resource name is not enough to uniquely identify an object. Please add resource kind and use it in `ListResourceEvents` method.

**Review by alexmt — CHANGES_REQUESTED** [review]

**Comment by alexmt** [comment]
LGTM

**Review by alexmt — APPROVED** [review]

**Review by jessesuen — APPROVED** [review]

**Pull request merged**